worldinfo.cpp

```diff
@@ -88,74 +107,130 @@ bool WorldInfo::parseFolder(const QDir &path)
-              continue;
+
+      // "file/<packname>.zip" -> zipped
+      if (dp.endsWith(".zip")) {
+        QStringRef pack_name(&dp, 5, dp.size()-5-4);
+        QString    zip_file = folder.path() + "/datapacks/" + pack_name + ".zip";
+        ZipReader zip(zip_file);
+        if (zip.open()) {
+          for (auto & file: zip.getFileList()) {
+
+            // name of datapack
+            // -> ./pack.mcmeta
+            if (file == "pack.mcmeta") {
+              QJsonDocument json_doc = QJsonDocument::fromJson(zip.get(file));
+              if (!json_doc.isNull())
+                parsePackMcmeta(json_doc, zip_file, true, true);
             }
-            QJsonObject json = json_doc.object();
-            // now 'json' should contain the Dimension description
-            QFileInfo f(file);
-            QString dim_name = f.baseName();
-            DimensionInfo dim;
-            dim.path = "./dimensions/" + pack_name + "/" + dim_name;
-            dim.name = pack_name + ":" + dim_name;
-
-            if (json.contains("type")) {
-              dim.id = json.value("type").toString();
-              parseDimensionType(dim, dim.id);
-            } else {
-              // mandatory field, in case it is missing -> silently try next one
-              continue;
+
+            // custom Dimensions
+            // ./data/<packname>/dimension/<dimensions>.json
+            if (file.startsWith("data/" + pack_name + "/dimension/") &&
+                file.endsWith(".json")) {
+              QJsonDocument json_doc = QJsonDocument::fromJson(zip.get(file));
+              if (!json_doc.isNull())
+                parseDimension(json_doc, pack_name.toString(), QFileInfo(file).baseName());
             }
 
-            // store it in list
-            dimensions.append(dim);
           }
+          zip.close();
+        }
+        continue;
+      }
+
+      // "file/<packname>" -> unzipped
+      {
+        QStringRef pack_name(&dp, 5, dp.size()-5);
+        QString    pack_path(folder.path() + "/datapacks/" + pack_name + "/");
+        // name of datapack
+        // -> ./pack.mcmeta
+        QFile f(pack_path + "pack.mcmeta");
+        if (f.open(QIODevice::ReadOnly)) {
+          QByteArray data = f.readAll();
+          f.close();
+          QJsonDocument json_doc = QJsonDocument::fromJson(data);
+          if (!json_doc.isNull())
+            parsePackMcmeta(json_doc, pack_path, true, false);
         }
 
-        zip.close();
+        // custom Dimensions
+        // ./data/<packname>/dimension/<dimensions>.json
+        // -> todo
+
       }
     }
   }
+}
 
-  return true;
+
+bool WorldInfo::parsePackMcmeta(const QJsonDocument & json_doc, const QString path, bool enabled, bool zipped)
+{
+  QJsonObject json = json_doc.object();
+  if (json.contains("pack")) {
+    QJsonObject pack =  json.value("pack").toObject();
+    if (pack.contains("description")) {
+      DatapackInfo dp;
+      dp.name    = pack.value("description").toString();
```

Comment:
Is `description` the right field to consider an identifier? I'd guess that this is the user-facing description of the pack, rather than an identifier.

Reply:
You might be right here ... I have to think about it, and look into some more datapacks.
There is some structure defining the namespace, we should use that.

Reply:
https://minecraft.wiki/w/Data_pack#Folder_structure -> [show]

Reply:
There is structure, but not a single field that would serve as a unique identifier. Nothing we should be basing detection upon.

Reply:
The namespace of a mod has to be unique and can not be changed without breaking the mod.

Reply:
#372